controllers/chat/sendSignedMessage.js

```diff
@@ -0,0 +1,80 @@
+const Validator = require('fastest-validator');
+const {StreamChat} = require('stream-chat');
+const {responseError, responseSuccess} = require('../../utils/Responses');
+const {CHANNEL_TYPE} = require('../../helpers/constants');
+
+const v = new Validator();
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ */
+const sendSignedMesage = async (req, res) => {
+  const schema = {
+    channelId: 'string|empty:false',
+    message: 'string|empty:false',
+    channelType: {
+      type: 'enum',
+      values: Object.values(CHANNEL_TYPE),
+      empty: false
+    }
+  };
+  const validated = v.validate(req.body, schema);
+  if (validated.length)
+    return res.status(403).json({
+      message: 'Error validation',
+      error: validated
+    });
+
+  const {channelId, message, channelType} = req.body;
+
+  let channelTypeDef;
+
+  switch (channelType) {
+    case CHANNEL_TYPE.CHAT:
+    case CHANNEL_TYPE.ANONYMOUS:
+      channelTypeDef = 'messaging';
+      break;
+
+    case CHANNEL_TYPE.GROUP:
+      channelTypeDef = 'group';
+      break;
+
+    default:
+      return res.status(403).json({
+        message: 'Error validation',
+        error: 'Channel type not found'
+      });
+  }
+
+  const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+  try {
+    await client.connectUser({id: req.userId}, req.token);
+
+    const channel = client.channel(channelTypeDef, channelId);
+
+    const createdChannel = await channel.create();
+
+    if (createdChannel?.channel?.is_channel_blocked)
+      return res.status(403).json(responseError('Channel is blocked'));
+
+    const chat = await channel.sendMessage({
+      user_id: req.userId,
+      text: message,
+      ...req.body
+    });
+
+    await client.disconnectUser();
+    return res.status(200).json(responseSuccess('sent', chat));
+  } catch (error) {
+    await client.disconnectUser();
+    return res.status(error.statusCode ?? error.status ?? 500).json({
+      status: 'error',
+      code: error.statusCode ?? error.status ?? 500,
+      message: error.message
+    });
+  }
+};
+
+module.exports = sendSignedMesage;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `sendSignedMesage` function seems to be well implemented. It validates the input using the `fastest-validator` library, connects to the Stream Chat API to send the message, handles errors, and returns appropriate responses. However, there's a potential issue with the way you're handling the request body in line 65. You're spreading the entire request body into the `sendMessage` method which could lead to unintended consequences if the request body contains additional fields that are not expected by the `sendMessage` method. 

To avoid this, it would be better to only pass the necessary fields to the `sendMessage` method.

```diff
-    const chat = await channel.sendMessage({
-      user_id: req.userId,
-      text: message,
-      ...req.body
-    });
+    const chat = await channel.sendMessage({
+      user_id: req.userId,
+      text: message
+    });
```

Also, consider renaming the function from `sendSignedMesage` to `sendSignedMessage`, as it appears to be a typo.


<!-- This is an auto-generated comment by OSS CodeRabbit -->